flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/savepoint/SavepointSerializers.java

```diff
@@ -35,7 +35,7 @@
 	private static final Map<Integer, SavepointSerializer<?>> SERIALIZERS = new HashMap<>(2);
 
 	static {
-		SERIALIZERS.put(SavepointV1.VERSION, SavepointV1Serializer.INSTANCE);
+		SERIALIZERS.put(SavepointV1Serializer.VERSION, SavepointV1Serializer.INSTANCE);
 		SERIALIZERS.put(SavepointV2.VERSION, SavepointV2Serializer.INSTANCE);
```

Comment:
Seems we just keep the recent savepoint, I'm thinking can we replacing all `SavepointV2` with `Savepoint` directly, after the replacement, we just have one `Savepoint` class, no `SavepointVX` class, and update the `VERSION` property in `Savepoint` when upgrading.